libmobilecoin/src/transaction.rs

```diff
@@ -18,22 +18,42 @@ use mc_util_ffi::*;
 /* ==== TxOut ==== */
 
 #[repr(C)]
-pub struct McTxOutAmount<'a> {
+pub struct McTxOutAmount {
     /// 32-byte `CompressedCommitment`
-    commitment: FfiRefPtr<'a, McBuffer<'a>>,
     masked_value: u64,
 }
 
-impl<'a> TryFromFfi<&McTxOutAmount<'a>> for Amount {
-    type Error = LibMcError;
+/// # Preconditions
+///
+/// * `view_private_key` - must be a valid 32-byte Ristretto-format scalar.
+#[no_mangle]
+pub extern "C" fn mc_tx_out_reconstruct_commitment(
+    tx_out_amount: FfiRefPtr<McTxOutAmount>,
+    tx_out_public_key: FfiRefPtr<McBuffer>,
+    view_private_key: FfiRefPtr<McBuffer>,
+    out_tx_out_commitment: FfiMutPtr<McMutableBuffer>,
+    out_error: FfiOptMutPtr<FfiOptOwnedPtr<McError>>,
```

Comment:
I think what I would suggest is, pass the `tx_out_commitment_crc32` as an argument to this perhaps?

Then you can compute the crc32 of the commitment bytes that this function produces, and confirm that they match, or return an error if they don't?

Reply:
That would be simpler than trying to do it in the other function I think, sorry for the misguided advice earlier

Reply:
ya that makes sense. I won't have time to implement and test that for this PR because it involves a decent amount of Swift changes. I will create a backlog ticket to add that functionality.